gensim/models/flsamodel.py

```diff
@@ -359,7 +392,8 @@ def _create_sparse_global_term_weights(
                 word_to_index,
             )
         else:
-            raise ValueError('Invalid word weighting method')
+            raise ValueError(f'Unsupported word_weighting {word_weighting}')
+        #ERijck: This ValueError can be omitted, as it is an internal method. Do you agree?
```

Comment:
Up to you. In general, input validation should only happen at public API points, where user sends in some data. 

And even there, the "pythonic" way is to rely on duck typing, and not require specific classes or methods.